components/financial_assistance/app/controllers/financial_assistance/deductions_controller.rb

```diff
@@ -66,8 +68,9 @@ def destroy
 
     def format_date(params)
       return if params[:deduction].blank?
-      params[:deduction][:start_on] = Date.strptime(params[:deduction][:start_on].to_s, "%m/%d/%Y")
-      params[:deduction][:end_on] = Date.strptime(params[:deduction][:end_on].to_s, "%m/%d/%Y") if params[:deduction][:end_on].present?
+      date_format = @bs4 ? "%Y-%m-%d" : "%m/%d/%Y"
```

Comment:
using th bs4 conditional makes it dependant on the layout rather than the actual format of the date, which is fine for now, but when it is on by default will be a bit off. Instead, I would check for the format of the param using a regex, so:
`date_format = params[:benefit][:start_on].match(/\d{4}-\d{2}-\d{2}/) ? "%Y-%m-%d" : "%m/%d/%Y"`

Reply:
fixed this here, and also some other controllers from other recent PRs of mine